Lib/test/test_urllib2.py

```diff
@@ -1304,6 +1304,8 @@ def test_redirect_fragment(self):
         fp = o.open('http://www.example.com')
         self.assertEqual(fp.geturl(), redirected_url.strip())
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```

Comment:
There are two tests failed on my computer, so I marked them `expectedFailure`:
```
======================================================================
FAIL: test_redirect_encoding (test.test_urllib2.HandlerTests) [b'/spaced path/']
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../vm/pylib-crate/Lib/test/test_urllib2.py", line 1358, in test_redirect_encoding
    self.assertTrue(request.startswith(expected), repr(request))
AssertionError: False is not true : b'GET http://example.com/spaced%20path/ HTTP/1.1\r\nAccept-Encoding: identity\r\nHost: example.com\r\nUser-Agent: Python-urllib/3.9\r\nConnection: close\r\n\r\n'

======================================================================
FAIL: test_redirect_encoding (test.test_urllib2.HandlerTests) [b'/?p\xc3\xa5-dansk']
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../vm/pylib-crate/Lib/test/test_urllib2.py", line 1358, in test_redirect_encoding
    self.assertTrue(request.startswith(expected), repr(request))
AssertionError: False is not true : b'GET http://example.com/?p%C3%A5-dansk HTTP/1.1\r\nAccept-Encoding: identity\r\nHost: example.com\r\nUser-Agent: Python-urllib/3.9\r\nConnection: close\r\n\r\n'
```
But it seems to pass on the GitHub CI.

My OS is Ubuntu 20.04, and the CPU is 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz.

My rustc version is 1.53.0, and the test command is
```
cargo run --release -- -m test -j8 -v
```

Is there anyone who can reproduce this error?